Source/WebCore/PAL/pal/graphics/WebGPU/Impl/WebGPUComputePipelineImpl.cpp

```diff
@@ -47,9 +47,8 @@ ComputePipelineImpl::~ComputePipelineImpl()
 
 Ref<BindGroupLayout> ComputePipelineImpl::getBindGroupLayout(uint32_t index)
 {
-    return m_bindGroupLayouts.ensure(index, [this, index] {
-        return BindGroupLayoutImpl::create(wgpuComputePipelineGetBindGroupLayout(m_backing, index), m_convertToBackingContext);
-    }).iterator->value;
+    // "A new GPUBindGroupLayout wrapper is returned each time"
```

Comment:
Is this change really necessary when the only writable member in GPUBindGroupLayout is label?

Can we return the same Ref<BindGroupLayout> inside a new GPUBindGroupLayout to satisfy the specification? Or is that not sufficient as the label will be different and one would expect the label of a GPUBindGroupLayout to go through to the entire device timeline

Reply:
I think it's up to us. It's a tradeoff: on one hand, fewer objects / allocations; on the other hand, labels will behave in a sensible manner.

I think we should start with the sensible labels first, and view this as a potential optimization opportunity when we get to the point of trying to optimize things.